source/visionEnhancementProviders/NVDAHighlighter.py

```diff
@@ -316,50 +316,57 @@ def onPanelActivated(self):
 		self.lastControl = self.optionsText
 
 	def _updateEnabledState(self):
-		settings = self._getSettingsStorage()
+		settingsStorage = self._getSettingsStorage()
 		settingsToTriggerActivation = [
-			settings.highlightBrowseMode,
-			settings.highlightFocus,
-			settings.highlightNavigator,
+			settingsStorage.highlightBrowseMode,
+			settingsStorage.highlightFocus,
+			settingsStorage.highlightNavigator,
 		]
-		if any(settingsToTriggerActivation):
-			if all(settingsToTriggerActivation):
-				self._enabledCheckbox.Set3StateValue(wx.CHK_CHECKED)
-			else:
-				self._enabledCheckbox.Set3StateValue(wx.CHK_UNDETERMINED)
-			self._ensureEnableState(True)
+		isAnyEnabled = any(settingsToTriggerActivation)
+		if all(settingsToTriggerActivation):
+			self._enabledCheckbox.Set3StateValue(wx.CHK_CHECKED)
+		elif isAnyEnabled:
+			self._enabledCheckbox.Set3StateValue(wx.CHK_UNDETERMINED)
 		else:
 			self._enabledCheckbox.Set3StateValue(wx.CHK_UNCHECKED)
-			self._ensureEnableState(False)
 
-	def _ensureEnableState(self, shouldBeEnabled: bool):
+		if not self._ensureEnableState(isAnyEnabled) and isAnyEnabled:
+			self._onEnableFailure()

```

Comment:
We have to do this here (`self._updateEnabledState()`) because it is called at the end of `makeSettings`. Though maybe it shouldn't be called there. Really, `makeSettings` should just be making the GUI reflect the current reality, though it's hard to know what to do when the config doesn't match the current state. This is also done in `_onCheckEvent`